apps/site/components/Common/Partners/PartnersIconList/index.tsx

```diff
@@ -0,0 +1,69 @@
+'use client';
+
+import { useEffect, useRef, useState } from 'react';
+import type { FC } from 'react';
+
+import { ICON_PARTNERS } from '#site/next.partners.constants';
+import type { PartnerCategory, Partners } from '#site/types';
+
+import PartnerIcon from '../PartnerIcon';
+import style from './index.module.css';
+import { randomPartnerList } from '../utils';
+
+type PartnersIconListProps = {
+  maxLength?: number;
+  categories?: PartnerCategory;
+};
+
+const PartnersIconList: FC<PartnersIconListProps> = ({
+  maxLength = 6,
+  categories,
+}) => {
+  const initialRenderer = useRef(true);
+
+  const [seedList, setSeedList] = useState<Array<Partners>>(
+    ICON_PARTNERS.slice(0, maxLength)
+  );
+
+  useEffect(() => {
```

Comment:
Could also be extracted to a shared hook similar to the other useEffect (maybe just one hook with shared functionality?)